### ISSUE TITLE
Release - June 2018- Target RC date: 2018-06-04 - name: 0.15.0

### ISSUE THREAD

**Issue by dslomov** [bazelbuild/bazel#5057]

**Comment by dslomov** [comment]
@laurentlb lmk if you are not available on these dates

**Comment by laurentlb** [comment]
We have a RC for Bazel 0.15: https://groups.google.com/forum/#!topic/bazel-discuss/v4f-wMyLyOo

4 projects are failing (https://buildkite.com/bazel/bazel-with-downstream-projects-bazel/builds/287):

* Bazel Remote Execution. Also failing with latest release.
* rules_python. There has been an open bug for a while: https://github.com/bazelbuild/rules_python/issues/98
* TensorFlow Serving (ubuntu 14.04). Also failing with latest release.
* TensorFlow Serving (ubuntu 16.04). Also failing with latest release.

**Comment by aehlig** [comment]
#5329 is a duplicate of #5323 but the latter is not yet fixed. Mentioning to keep track of the blocking issues.

**Comment by petemounce** [comment]
rc1 is on chocolatey.

**Comment by dslomov** [comment]
#5336 is a release blocker, let's cherry-pick https://github.com/bazelbuild/bazel/commit/4b80f2455e7e49a95f3a4c9102a67a57dad52207

**Comment by dslomov** [comment]
Two more cherrypicks, the same that went into 0.14.1 (#5340):
 * 4b80f2455e7e49a95f3a4c9102a67a57dad52207 
 * 6b1635279e8b33dc1ac505ac81825e38f8797a14

**Comment by laurentlb** [comment]
rc2 with the two cherrypicks: https://releases.bazel.build/0.15.0/rc2/index.html

**Comment by dslomov** [comment]
How about https://github.com/bazelbuild/bazel/commit/4ec0a7524913ab2c4641368e3f8c09b347351a08? It fixes a regression introduced in 0.14.

**Comment by laurentlb** [comment]
rc3 was created 7 days ago: https://releases.bazel.build/0.15.0/rc3/index.html
It includes the 3 cherry-picks requested in this thread.

**Comment by RNabel** [comment]
The release notes state: `the experimental_remote_spawn_cache flag is now enabled by default`. If this flag is experimental, why is it enabled by default? Are there any more docs on what the impact of this flag are?

**Comment by laurentlb** [comment]
This comes from this commit: https://github.com/bazelbuild/bazel/commit/cdf118a607ab179807cd39d0cafdb551859d1cd1#diff-b5fe53cc60be2450f4c7c621f5931370

@buchgr Can you clarify this? It would be nice to provide more context in the release announcement (https://docs.google.com/document/d/1wDvulLlj4NAlPZamdlEVFORks3YXJonCjyuQMUQEmB0/edit).

Thanks!

**Comment by buchgr** [comment]
@RNabel 

I can now see that the wording is unfortunate. We graduated the flag to non-experimental by enabling it by default. It's in fact now a no-op, when using the remote cache. We kept it in for the 0.15.0 in order to not break any builds.

**Comment by buchgr** [comment]
released! @vbatts @petemounce @ilovezfs

**Comment by ilovezfs** [comment]
https://github.com/Homebrew/homebrew-core/pull/29447

**Comment by ilovezfs** [comment]
0.15.0 shipped in Homebrew.

**Comment by vbatts** [comment]
fedora and centos build is kicked off https://copr.fedorainfracloud.org/coprs/vbatts/bazel/build/771133/
Also, I recently started stripping the binary, but was having issues isolating just the debug info from the binary so that could be available for folks interested. Not sure the issue there. But for now, the centos and fedora builds are stripped to 1.5Mb from their 100.5Mb original size :-)

**Comment by buchgr** [comment]
@vbatts tell me more? how did you accomplish this?

**Comment by vbatts** [comment]
email sent

**Comment by evantorrie** [comment]
@vbatts Did you test the resulting build?
After installing the stripped versions from vbatts/bazel (including 0.14.1-2), I get the following error on Fedora 28

```
[evant@learncollege-dl envoy]$ bazel
file is invalid or corrupted (missing end of central directory record)
Cannot find central directory
FATAL: Failed to open Bazel as a zip file: (error: 5): Input/output error
```
I'm pretty sure the stripping is stripping out the packaged internal binary

**Comment by petemounce** [comment]
Chocolatey is pushed.

**Comment by vbatts** [comment]
@evantorrie I see that. I must have not hit that on the 0.14.1-1 -> 0.14.1-2 upgrade locally because the extracted cache seemed fresh? not sure.
I have an updated/fixed build in the works for 0.15.0 https://copr.fedorainfracloud.org/coprs/vbatts/bazel/build/771662/

**Comment by dslomov** [comment]
0.15.0 is done, closing this.
